### PR TITLE
remote: fetch, adds the prune option.

### LINKED ISSUE #316: Support prune option in Fetch method
## Feature

Currently, it is not possible to `fetch` from a remote and to remove the deleted remote branches at the same time. This is currently possible with the `git fetch` command by using the `--prune` option as described here https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt---prune.

As suggested in https://github.com/src-d/go-git/issues/624 a `bool` could be added to the `FetchOptions` struct.

As a side note, the above issue has been closed without a proper fix because the linked PR only adds the `Prune` option to the `PushOptions` struct which is not the same.

## Workaround

There is currently no workaround or at least none of I'm aware of.

### PR DESCRIPTION
**Pull request by juliens**
This PR adds `prune` option to the `Fetch` command.

Fixes #316

### CODE REVIEW

File: remote_test.go [hunk 1]
```diff
@@ -1134,3 +1134,117 @@ func (s *RemoteSuite) TestPushRequireRemoteRefs(c *C) {
...
+	AssertReferences(c, rSave, map[string]string{
+		"refs/tags/v1": ref.Hash().String(),
+	})
+
+	err = rSave.Fetch(&FetchOptions{Prune: true, RefSpecs: []config.RefSpec{"refs/tags/*:refs/tags/*"}})
+	c.Assert(err, IsNil)
+
+	_, err = rSave.Reference("refs/tags/v1", true)
+	c.Assert(err, ErrorMatches, "reference not found")
+}
```
Comment (pjbgf) [inline remote_test.go:1250]:
```suggestion
}

```

**Review by pjbgf — COMMENTED** [review]
LGTM

@juliens would you mind rebasing this PR so we can get it merged?

**Comment by juliens** [comment]
Done ;)

**Comment by pjbgf** [comment]
@juliens thank you for the quick turn around. One small nit, for the PR validation to pass you will need to use `git:` as the prefix for the commit message. We generally use `*:` or `git:` for changes on the main package.

**Comment by juliens** [comment]
I thought that `remote:` would work. I changed for `*:`, tell me if you thing I should have use another one.

**Review by pjbgf — APPROVED** [review]
@juliens thank you for working on this. :bow:

**Pull request merged**